components/Item/Item.js

```diff
@@ -79,7 +79,7 @@ const Item = props => (
       }
     `}</style>
     <li className="product_card">
-      <Link as={`/product/${props.id}`} href={`/product?id=${props.id}`}>
```

Comment:
@Nshul I think we should have `as` attribute for masking our routes. If we look at other sites like Myntra for inspiration they have the `item.name` in the URL rather than the id.